definition_validator/validator.cc

```diff
@@ -563,7 +558,32 @@ void validateCompatibleOverride(const core::Context ctx, const ast::ExpressionPt
                                superReturn.show(ctx));
                 e.addErrorNote("A method's return type must be a subtype of the return type on the super method.");
                 e.addErrorSections(move(errorDetailsCollector));
-                e.maybeAddAutocorrect(constructAllowIncompatibleAutocorrect(ctx, tree, methodDef, reportedAutocorrect));
+                e.maybeAddAutocorrect(
+                    constructAllowIncompatibleAutocorrect(ctx, tree, methodDef, "true", reportedAutocorrect));
+            }
+        }
+    }
+
+    // Check method visibility
+
+    if (!method.data(ctx)->flags.allowIncompatibleOverrideVisibility &&
+        ((method.data(ctx)->flags.isPrivate &&
+          (superMethod.data(ctx)->flags.isProtected || superMethod.data(ctx)->isMethodPublic())) ||
+         (method.data(ctx)->flags.isProtected && superMethod.data(ctx)->isMethodPublic()))) {
+        if (auto e = ctx.beginError(methodDef.declLoc, core::errors::Resolver::BadMethodOverride)) {
+            auto modifier = method.data(ctx)->flags.isPrivate ? "private" : "protected";
+            e.setHeader("Method `{}` is {} in `{}` but not in `{}`", method.data(ctx)->name.show(ctx), modifier,
+                        method.data(ctx)->owner.show(ctx), superMethod.data(ctx)->owner.show(ctx));
+            e.addErrorLine(superMethod.data(ctx)->loc(), "Base method defined here");
+
+            auto len = method.data(ctx)->flags.isPrivate ? 7 : 9;
+            auto loc = ctx.locAt(methodDef.declLoc).adjustLen(ctx, -(len + 1), len);
+            if (ctx.state.suggestUnsafe) {
+                // We don't want to pass `repeatedAutocorrect` here, because we want to overwrite
+                // with `allow_incompatible: true` if there are multiple errors.
```

Comment:
Maybe this reference to repeatedAutocorrect is from an old, work-in-progress implementation?